lab/03-testing.md

```diff
@@ -2,52 +2,73 @@
 
 # Testing
 
-We need to test that our django application works as expected. We have provided some tests that ensure that you can visit all of the urls and that the data is being returned as expected.
+We need to test that our Django application works as expected. We have provided some tests that ensure that you can visit all of the urls and that the data is being returned as expected.
 
-## Testing via Django Tests
-The first set of tests we'll run are the Django tests. These tests are run inside of the container and are a great way to ensure that your application is working as expected. Django does a fantastic job of setup and teardown of test resources 
+## Installing the prerequisites
 
+The Dev Container already has the production requirements for the Django app installed,
+but it doesn't have the development requirements yet: the linting and testing packages.
 
-navigate to the `demo-code` folder and run the following command:
+Install the development requirements via pip:
 
 > **:computer: Try It!**
 >```shell
+> python -m pip install -r requirements-dev.txt
+> ```
+
+## Testing via Django Tests
+
+The first set of tests we'll run are the Django tests. These tests are run inside of the container and are a great way to ensure that your application is working as expected. Django does a fantastic job of setup and teardown of test resources.
+
+Navigate to the `demo-code` folder and run the `test` command:
+
+> **:computer: Try It!**
+> ```shell
 > cd demo-code
-> python manage.py relecloud.test
+> python manage.py collectstatic
+> python manage.py test relecloud.tests
 > ```
 
 https://user-images.githubusercontent.com/8632637/229249044-7c8b65d1-3fdc-46a1-9c72-09519452500d.mp4
 
 
-## Testing the frontend of the Application with Playwright
+## Testing the frontend of the application with Playwright
 
-There are somethings you can't test for without loading the website. For that we'll use [Playwright](https://playwright.dev/).
+There are some things you can't test without loading the website. For that we'll use [Playwright](https://playwright.dev/), a browser automation library that allows us to run tests against our website. We've included tests that will navigate across multiple pages of the website, similar to how a user would.
 
-Playwright is a browser automation library that allows us to run tests against our website. We've included tests that will check for specific values are correct and that the websites functions as expected.
+First, install Playwright and its dependencies, the headless browser builds. ⚠️ If you're on an M1 Mac, it's not possible to run the browsers from inside the Dev Container. In that case, either use GitHub Codespaces or a local virtual environment.
 
-We're using pytest with the pytest-playwright extension. To run your playright tests, run pytest.
+> **:computer: Try It**
+> ```shell
+> playwright install
+> playwright install-deps
```

Comment:
Ditto, could also go in devcontainer.json, but will slow down build time

Reply:
I can't remember how long it took but definitely would prefer slower (and questionably optional) build steps not be in the devcontainer.